From: Sebastian Wick <sebastian.wick@redhat.com>
Date: Fri, 4 Mar 2022 18:48:31 +0100
Subject: window: Check fullscreen status when window changes monitor

For wayland meta_window_move_to_monitor sends a configure to the client
without actually moving the window, yet and the
meta_display_queue_check_fullscreen call won't detect any changes.
Checking for fullscreen in meta_window_update_monitor fixes the problem
because it is called whenever the window actually changed the monitor it
is on.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2325>

Origin: https://gitlab.gnome.org/GNOME/mutter/-/commit/1760dcb0
---
 src/core/window.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/window.c b/src/core/window.c
index 2545a1b..ccc6a28 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -3744,6 +3744,8 @@ meta_window_update_monitor (MetaWindow                   *window,
       /* If we're changing monitors, we need to update the has_maximize_func flag,
        * as the working area has changed. */
       meta_window_recalc_features (window);
+
+      meta_display_queue_check_fullscreen (window->display);
     }
 }
 
